### ISSUE TITLE
After power off ESXi host, disk cannot be mounted to new node.

### ISSUE THREAD

**Issue by luomiao** [vmware/kubernetes#116]
Shanghai team reported the issue when testing with mongoDB.
Each node has multiple pods and each pod has their own persistent volume.
After power off one of the ESXi host, the node is restarted on another host.
However the disk cannot be mounted to other node.
Error from vCenter:
```
File system specific implementation of LookupAndOpen[file] failed
Failed to add disk 'scsi1:3'.
Cannot open the disk '/vmfs/volumes/vsan:5231ae302a9fb63a-fe0fadf567dd24cb/562ba158-d8a2-046f-6ba7-0050569ce285/k8s-test-cluster-1-dynamic-pvc-19e1e6d7-07cd-11e7-a07b-005056b281c6.vmdk' or one of the snapshot disks it depends on.
```
 
From kubernetes:
```
Failed to attach volume "pvc-1a79af4e-07cd-11e7-a07b-005056b281c6" on node "node6" with: Failed to add disk 'scsi1:0'.
Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "default"/"mongodb-shard-node01-472414307-qvl77". list of unattached/unmounted volumes=[vsphere-storage-mongoc01 vsphere-storage-db-rsp01 vsphere-storage-db-rss04 vsphere-storage-db-rsa0
```

Divyen tried to reproduce this error with a single wordpress pod and everything looks good.
The same error shows up for a while but disappears after the pod is rescheduled.

The next step we may want to try out the same yaml files to see if the mounting will be successful when there are multiple disks need to be unmounted/mounted.

**Comment by divyenpatel** [comment]
Just checked Shanghai team's VSAN Cluster, I see default VSAN Policy has `Number of failures to tolerate` is set to `1`.

If they are failing over one host. we should not encounter this issue. but if they are powering off couple of hosts from hardware console, then node VM's primary disk can become inaccessible and so master can never unmount and detach volume from that node.

**Comment by luomiao** [comment]
@divyenpatel 
Just verified that only one host is powered off at one time... So this might not be the reason of the error.

**Comment by divyenpatel** [comment]
@luomiao  

even if we power off one host after another with some delays, some nodes may become inaccessible after we power off second host, and can lead to this situation. as only two copies of the nodes disk is present.

For example.

VM node1 is deployed on host1.
VM node2 is deployed on host2.
VM node3 is deployed on host3.
VM master is deployed in host3.

node1's primary disk's copy is available on host1 and host2.
node2's primary disk's copy is available on host1 and host2.
node3's primary disk's copy is available on host3 and host2.
master's primary disk's copy is available on host4 and host3.

VM node1 has the pod and volume disk is attached to node1. 

**Power off host1**. assume node1 VM will be restarted on host3. 
and at the same time pod is scheduled to be recreated on node2 which is deployed on host2.
volume disk will be detached from node1 and attached to node2.

now **power off host2**, which has node2 and the pod.
assume node2 be restarted on host3 and pod is scheduled to be recreated on node3.

**node2's primary disk's copies were available only on host1 and host2. and so node2's disk will become inaccessible, but in VC node2 is still powered on VM - with volume disk attached.**

**even if pod is scheduled to be created on node3, cloud provider will never be able to attach hot volume disk to node3, as volume disk is already attached to powered on vm node2** 


For two host failover, we should change default VSAN profile and set “`Number of failures to tolerate`” to “`2`”. Also policy needs to be re-applied back to node VMs, so that appropriate copies of objects are created. if volumes were created before increasing FTT, policy needs to be re-applied to those volume disks.

Can we ask Shanghai team to increase FTT to two and then perform two host failover?

**Comment by luomiao** [comment]
We were able to reproduce this issue on testbed prepared by Shanghai team.
At the beginning one pod is running on node 4 with 4 volumes attached/mounted.
After we powered off the host ESXi where node 4 is running on, the Pod is reschedule to node 5.
However, only 2 disks are successfully attached/mounted to node 5.
The other 2 disks are still attached to node 4, with mount path empty:
```
root@node4 [ /var/lib/docker/containers ]# lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sdc      8:32   0   30G  0 disk
sdd      8:48   0   10G  0 disk
sde      8:64   0 15.6G  0 disk
|-sde1   8:65   0    2M  0 part
`-sde2   8:66   0    8G  0 part /
sr0     11:0    1 1024M  0 rom
```

The logs have a bunch of messages:
```
{"log":"W0324 03:48:33.700489     458 mount.go:136] could not determine device for path: \"/var/lib/kubelet/pods/403c8515-07cd-11e7-a07b-005056b281c6/volumes/kubernetes.io~vsphere-volume/pvc-1bd0299d-07cd-11e7-a07b-005056b281c6\"\n","stream":"stderr","time":"2017-03-24T03:48:33.700730434Z"}
{"log":"E0324 03:48:33.701679     458 reconciler.go:433] Could not construct volume information: directory /var/lib/kubelet/pods/403c8515-07cd-11e7-a07b-005056b281c6/volumes/kubernetes.io~vsphere-volume/pvc-1bd0299d-07cd-11e7-a07b-005056b281c6 is not mounted\n","stream":"stderr","time":"2017-03-24T03:48:33.701751806Z"}
```

and 

```
{"log":"E0324 03:48:34.604697     458 kubelet_volumes.go:110] Orphaned pod \"403c8515-07cd-11e7-a07b-005056b281c6\" found, but error \u003cnil\u003e occured during reading volume dir from disk\n","stream":"stderr","time":"2017-03-24T03:48:34.604854878Z"}
{"log":"E0324 03:48:34.604916     458 kubelet_volumes.go:110] Orphaned pod \"5454a242-0ed2-11e7-a07b-005056b281c6\" found, but error \u003cnil\u003e occured during reading volume dir from disk\n","stream":"stderr","time":"2017-03-24T03:48:34.60548334Z"}
{"log":"E0324 03:48:34.605127     458 kubelet_volumes.go:110] Orphaned pod \"647dbda9-0d8d-11e7-a07b-005056b281c6\" found, but error \u003cnil\u003e occured during reading volume dir from disk\n","stream":"stderr","time":"2017-03-24T03:48:34.605557595Z"}
{"log":"E0324 03:48:34.605270     458 kubelet_volumes.go:110] Orphaned pod \"ec836ab2-0f91-11e7-a07b-005056b281c6\" found, but error \u003cnil\u003e occured during reading volume dir from disk\n","stream":"stderr","time":"2017-03-24T03:48:34.605569314Z"}
```

**Comment by divyenpatel** [comment]
@luomiao, tried to reproduce issue with 10 Volume pod, unable to reproduce the issue.
All 10 Volumes got attached to the new node where POD is re-scheduled.


**Kubernetes Version I have is : v1.5.7-beta.0.2+50e04da70f6d4a**

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:40:50Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.7-beta.0.2+50e04da70f6d4a", GitCommit:"50e04da70f6d4a61648b41fcb94f3a2a28a28dec", GitTreeState:"clean", BuildDate:"2017-03-29T00:38:51Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```



**Created pod with 10 Volumes. pod placed on node1. All volumes attached to node1.**

```
$ kubectl create -f 10-disk-mysqlPod.yaml
service "wordpress-mysql" created
persistentvolumeclaim "mysql-pv-claim1" created
persistentvolumeclaim "mysql-pv-claim2" created
persistentvolumeclaim "mysql-pv-claim3" created
persistentvolumeclaim "mysql-pv-claim4" created
persistentvolumeclaim "mysql-pv-claim5" created
persistentvolumeclaim "mysql-pv-claim6" created
persistentvolumeclaim "mysql-pv-claim7" created
persistentvolumeclaim "mysql-pv-claim8" created
persistentvolumeclaim "mysql-pv-claim9" created
persistentvolumeclaim "mysql-pv-claim10" created
deployment "wordpress-mysql" created
```

```
$ kubectl get pvc
NAME               STATUS    VOLUME                                     CAPACITY   ACCESSMODES   AGE
mysql-pv-claim1    Bound     pvc-22c1f749-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim10   Bound     pvc-22f1155b-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim2    Bound     pvc-22c7b37a-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim3    Bound     pvc-22cc8560-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim4    Bound     pvc-22d1b151-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim5    Bound     pvc-22d6b2eb-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim6    Bound     pvc-22dbf7f2-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim7    Bound     pvc-22e12ac7-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim8    Bound     pvc-22e6dbbc-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
mysql-pv-claim9    Bound     pvc-22ec090e-14ab-11e7-ae77-005056bdb770   2Gi        RWO           5m
$
```

```
$ kubectl describe pod wordpress-mysql-4011518514-zmfqj
Name:		wordpress-mysql-4011518514-zmfqj
Namespace:	default
Node:		node1/10.160.8.156
Start Time:	Wed, 29 Mar 2017 11:11:46 -0700
Labels:		app=wordpress
		pod-template-hash=4011518514
		tier=mysql
Status:		Running
IP:		172.1.15.2
Controllers:	ReplicaSet/wordpress-mysql-4011518514
Containers:
  mysql:
    Container ID:	docker://e6db586f6247ba7fa76a84b0ba460d78df9e009a4072d6051f7b5c71d3b3695f
    Image:		mysql:5.6
    Image ID:		docker://sha256:68399df75c596686ace0c41106d1506965b15d54dd48f8599ad94ef589e0fc3e
    Port:		3306/TCP
    State:		Running
      Started:		Wed, 29 Mar 2017 11:12:33 -0700
    Ready:		True
    Restart Count:	0
    Volume Mounts:
      /var/lib/mysql from mysql-persistent-storage1 (rw)
      /var/lib/mysql10 from mysql-persistent-storage10 (rw)
      /var/lib/mysql2 from mysql-persistent-storage2 (rw)
      /var/lib/mysql3 from mysql-persistent-storage3 (rw)
      /var/lib/mysql4 from mysql-persistent-storage4 (rw)
      /var/lib/mysql5 from mysql-persistent-storage5 (rw)
      /var/lib/mysql6 from mysql-persistent-storage6 (rw)
      /var/lib/mysql7 from mysql-persistent-storage7 (rw)
      /var/lib/mysql8 from mysql-persistent-storage8 (rw)
      /var/lib/mysql9 from mysql-persistent-storage9 (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-6fd49 (ro)
    Environment Variables:
      MYSQL_ROOT_PASSWORD:	mysqlpassword
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  mysql-persistent-storage1:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim1
    ReadOnly:	false
  mysql-persistent-storage2:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim2
    ReadOnly:	false
  mysql-persistent-storage3:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim3
    ReadOnly:	false
  mysql-persistent-storage4:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim4
    ReadOnly:	false
  mysql-persistent-storage5:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim5
    ReadOnly:	false
  mysql-persistent-storage6:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim6
    ReadOnly:	false
  mysql-persistent-storage7:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim7
    ReadOnly:	false
  mysql-persistent-storage8:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim8
    ReadOnly:	false
  mysql-persistent-storage9:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim9
    ReadOnly:	false
  mysql-persistent-storage10:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim10
    ReadOnly:	false
  default-token-6fd49:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-6fd49
QoS Class:	BestEffort
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath		Type		Reason			Message
  ---------	--------	-----	----			-------------		--------	------			-------
  6m		6m		5	{default-scheduler }				Warning		FailedScheduling	[SchedulerPredicates failed due to PersistentVolumeClaim is not bound: "mysql-pv-claim1", which is unexpected., SchedulerPredicates failed due to PersistentVolumeClaim is not bound: "mysql-pv-claim1", which is unexpected., SchedulerPredicates failed due to PersistentVolumeClaim is not bound: "mysql-pv-claim1", which is unexpected., SchedulerPredicates failed due to PersistentVolumeClaim is not bound: "mysql-pv-claim1", which is unexpected.]
  6m		6m		1	{default-scheduler }				Normal		Scheduled		Successfully assigned wordpress-mysql-4011518514-zmfqj to node1
  5m		5m		1	{kubelet node1}		spec.containers{mysql}	Normal		Pulling			pulling image "mysql:5.6"
  5m		5m		1	{kubelet node1}		spec.containers{mysql}	Normal		Pulled			Successfully pulled image "mysql:5.6"
  5m		5m		1	{kubelet node1}		spec.containers{mysql}	Normal		Created			Created container with docker id e6db586f6247; Security:[seccomp=unconfined]
  5m		5m		1	{kubelet node1}		spec.containers{mysql}	Normal		Started			Started container with docker id e6db586f6247
```

![node1-with-10-volumes-attached](https://cloud.githubusercontent.com/assets/22985595/24470404/aacbb37e-1473-11e7-916c-27bf56f3163c.jpg)

**Powered off Node1's Host, and POD is scheduled to start on node3.**


```
$ kubectl get pod
NAME                               READY     STATUS              RESTARTS   AGE
wordpress-mysql-4011518514-7n8dc   0/1       ContainerCreating   0          20s
```

```
$ kubectl describe pod wordpress-mysql-4011518514-7n8dc
Name:		wordpress-mysql-4011518514-7n8dc
Namespace:	default
Node:		node3/10.160.6.184
Start Time:	Wed, 29 Mar 2017 11:22:42 -0700
Labels:		app=wordpress
		pod-template-hash=4011518514
		tier=mysql
Status:		Running
IP:		172.1.23.3
Controllers:	ReplicaSet/wordpress-mysql-4011518514
Containers:
  mysql:
    Container ID:	docker://47cf7554025f533ecefe3b88e706c714056b2dceec021e1d1262f610f5ed1a02
    Image:		mysql:5.6
    Image ID:		docker://sha256:68399df75c596686ace0c41106d1506965b15d54dd48f8599ad94ef589e0fc3e
    Port:		3306/TCP
    State:		Running
      Started:		Wed, 29 Mar 2017 11:27:56 -0700
    Ready:		True
    Restart Count:	0
    Volume Mounts:
      /var/lib/mysql from mysql-persistent-storage1 (rw)
      /var/lib/mysql10 from mysql-persistent-storage10 (rw)
      /var/lib/mysql2 from mysql-persistent-storage2 (rw)
      /var/lib/mysql3 from mysql-persistent-storage3 (rw)
      /var/lib/mysql4 from mysql-persistent-storage4 (rw)
      /var/lib/mysql5 from mysql-persistent-storage5 (rw)
      /var/lib/mysql6 from mysql-persistent-storage6 (rw)
      /var/lib/mysql7 from mysql-persistent-storage7 (rw)
      /var/lib/mysql8 from mysql-persistent-storage8 (rw)
      /var/lib/mysql9 from mysql-persistent-storage9 (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-6fd49 (ro)
    Environment Variables:
      MYSQL_ROOT_PASSWORD:	mysqlpassword
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  mysql-persistent-storage1:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim1
    ReadOnly:	false
  mysql-persistent-storage2:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim2
    ReadOnly:	false
  mysql-persistent-storage3:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim3
    ReadOnly:	false
  mysql-persistent-storage4:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim4
    ReadOnly:	false
  mysql-persistent-storage5:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim5
    ReadOnly:	false
  mysql-persistent-storage6:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim6
    ReadOnly:	false
  mysql-persistent-storage7:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim7
    ReadOnly:	false
  mysql-persistent-storage8:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim8
    ReadOnly:	false
  mysql-persistent-storage9:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim9
    ReadOnly:	false
  mysql-persistent-storage10:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	mysql-pv-claim10
    ReadOnly:	false
  default-token-6fd49:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-6fd49
QoS Class:	BestEffort
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath		Type		Reason		Message
  ---------	--------	-----	----			-------------		--------	------		-------
  7m		7m		1	{default-scheduler }				Normal		Scheduled	Successfully assigned wordpress-mysql-4011518514-7n8dc to node3
  6m		6m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22cc8560-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  6m		6m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22f1155b-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  5m		5m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22c7b37a-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  5m		5m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22e12ac7-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  4m		4m		1	{kubelet node3}					Warning		FailedMount	Unable to mount volumes for pod "wordpress-mysql-4011518514-7n8dc_default(b2c5b116-14ac-11e7-ae77-005056bdb770)": timeout expired waiting for volumes to attach/mount for pod "default"/"wordpress-mysql-4011518514-7n8dc". list of unattached/unmounted volumes=[mysql-persistent-storage1 mysql-persistent-storage2 mysql-persistent-storage3 mysql-persistent-storage4 mysql-persistent-storage5 mysql-persistent-storage6 mysql-persistent-storage7 mysql-persistent-storage8 mysql-persistent-storage9 mysql-persistent-storage10]
  4m		4m		1	{kubelet node3}					Warning		FailedSync	Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "default"/"wordpress-mysql-4011518514-7n8dc". list of unattached/unmounted volumes=[mysql-persistent-storage1 mysql-persistent-storage2 mysql-persistent-storage3 mysql-persistent-storage4 mysql-persistent-storage5 mysql-persistent-storage6 mysql-persistent-storage7 mysql-persistent-storage8 mysql-persistent-storage9 mysql-persistent-storage10]
  4m		4m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22d1b151-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  4m		4m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22d6b2eb-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  3m		3m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22dbf7f2-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  3m		3m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22ec090e-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  3m		3m		1	{controller-manager }				Warning		FailedMount	Failed to attach volume "pvc-22e6dbbc-14ab-11e7-ae77-005056bdb770" on node "node3" with: Failed to add disk 'scsi1:0'.
  2m		2m		1	{controller-manager }				Warning		FailedMount	(events with common reason combined)
  2m		2m		1	{kubelet node3}					Warning		FailedMount	Unable to mount volumes for pod "wordpress-mysql-4011518514-7n8dc_default(b2c5b116-14ac-11e7-ae77-005056bdb770)": timeout expired waiting for volumes to attach/mount for pod "default"/"wordpress-mysql-4011518514-7n8dc". list of unattached/unmounted volumes=[mysql-persistent-storage8 mysql-persistent-storage10]
  2m		2m		1	{kubelet node3}					Warning		FailedSync	Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "default"/"wordpress-mysql-4011518514-7n8dc". list of unattached/unmounted volumes=[mysql-persistent-storage8 mysql-persistent-storage10]
  1m		1m		1	{kubelet node3}		spec.containers{mysql}	Normal		Pulling		pulling image "mysql:5.6"
  1m		1m		1	{kubelet node3}		spec.containers{mysql}	Normal		Pulled		Successfully pulled image "mysql:5.6"
  1m		1m		1	{kubelet node3}		spec.containers{mysql}	Normal		Created		Created container with docker id 47cf7554025f; Security:[seccomp=unconfined]
  1m		1m		1	{kubelet node3}		spec.containers{mysql}	Normal		Started		Started container with docker id 47cf7554025f
```

```
$ kubectl exec -it wordpress-mysql-4011518514-7n8dc bin/sh
# df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay         7.8G  2.3G  5.1G  31% /
tmpfs           3.9G     0  3.9G   0% /dev
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/root       7.8G  2.3G  5.1G  31% /etc/hosts
/dev/sdg        2.0G  3.0M  1.8G   1% /var/lib/mysql5
/dev/sde        2.0G  3.0M  1.8G   1% /var/lib/mysql7
/dev/sdk        2.0G  119M  1.7G   7% /var/lib/mysql
/dev/sdd        2.0G  3.0M  1.8G   1% /var/lib/mysql2
/dev/sdj        2.0G  3.0M  1.8G   1% /var/lib/mysql8
/dev/sdi        2.0G  3.0M  1.8G   1% /var/lib/mysql9
/dev/sdc        2.0G  3.0M  1.8G   1% /var/lib/mysql10
/dev/sdb        2.0G  3.0M  1.8G   1% /var/lib/mysql3
/dev/sdh        2.0G  3.0M  1.8G   1% /var/lib/mysql6
/dev/sdf        2.0G  3.0M  1.8G   1% /var/lib/mysql4
tmpfs           3.9G   12K  3.9G   1% /run/secrets/kubernetes.io/serviceaccount
shm              64M     0   64M   0% /dev/shm
# 
```




**Verified All Volumes are now attached to node3.**

![node3-with-10-volumes-attached-tasks](https://cloud.githubusercontent.com/assets/22985595/24470585/46c1ddee-1474-11e7-9437-40110409ac31.jpg)

![node3-with-10-volumes-attached](https://cloud.githubusercontent.com/assets/22985595/24470589/4b34ccc4-1474-11e7-9167-ca3ab55cf710.jpg)

**Comment by divyenpatel** [comment]
@luomiao As we discussed, when I tried with higher capacity volume I am able to reproduce this issue. One of the 30 GB volume was left in the failed over node.

I am continuously seeing message on the failed over node.
`kubelet_volumes.go:110] Orphaned pod "60185498-14b4-11e7-ae77-005056bdb770" found, but error <nil> occured during reading volume dir from disk`

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_volumes.go#L110

```
# lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sdc      8:32   0   30G  0 disk 
sdd      8:48   0 15.6G  0 disk 
|-sdd1   8:49   0    2M  0 part 
`-sdd2   8:50   0    8G  0 part /
sr0     11:0    1 1024M  0 rom 
```

**Comment by luomiao** [comment]
Great that we finally caught this issue!
So a larger volume may need more time to get unmounted/detached.
We need to figure out what's the order when the pods/nodes getting rescheduled.
What's the order of unmounting volumes, killing old pod, scheduling new pod.

**Comment by divyenpatel** [comment]
Issue is producible on release 1.6 branch also. 
Observed this failure on build v1.6.3-beta.0.43+ab61393ee6b119

Created an issue on Kubernetes to get help from Kubernetes SIG storage.
https://github.com/kubernetes/kubernetes/issues/45464

**Comment by divyenpatel** [comment]
Merged and cherry picked to 1.6
